src/python/ensembl/brc4/runnable/core_server.py

```diff
@@ -55,8 +57,12 @@ def connect(self) -> None:
         )
 
     def set_database(self, db_name: str) -> None:
-        self._connector.database = db_name
-
+        try:
+            self._connector.database = db_name
+        except:
+            logging.exception(f"Unknown database ! DB:'{db_name}' not located on host {self.host}")
+            raise OperationalError(f"Unknown database!", None, None)
```

Comment:
It is advisable to only handle the exception you are interested in, and I would also suggest to just re-raise after the logging step:
```suggestion
        except mysql.connector.errors.ProgrammingError:
            logging.exception(f"Unknown database ! DB:'{db_name}' not located on host {self.host}")
            raise
```